### ISSUE TITLE
[openbmc] rspconfig - Support a way to send the ssh keys over to the BMC 

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3308]
Currently there's no REST command to upload a public key to the BMC. 

Rather than wait for this support, leverage the `rspconfig node sshcfg` option to use Expect for now to send over the ssh key so we can passwordless ssh to the BMC 

*Note* May need to create the `/root/.ssh` directory since that is missing from openbmc at the time of this writing

**Comment by whowutwut** [comment]
This issue should be put on hold until we get confirmation on which milestone https://github.com/openbmc/openbmc/issues/1517 will be targeted for.  Should know more next week.

**Comment by gurevichmark** [comment]
It appears that no REST API will be available in the near future, so for now will implement `scp`

**Comment by gurevichmark** [comment]
Thinking of using `sshpass` command that should already be there. If not, can be installed with `yum -y install sshpass`. The `sshpass` allows passing of password on the command line. `BMC_PW` will be extracted from xCAT `passwd` table

1. Create a `.ssh` directory on BMC: 
   `sshpass -p <BMC_PW> ssh <BMC_IP> "mkdir -p ~/.ssh"`

2. Copy `id_rsa.pub` file from MN to BMC: 
   `sshpass -p <BMC_PW> scp ~/.ssh/id_rsa.pub <BMC_IP> ~/.ssh/id_rsa.pub"`

3. Append contents of copied `id_rsa.pub` to `authorized_keys`: 
   `sshpass -p <BMC_PW> ssh <BMC_IP> "cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys"`

4. Remove copied `id_rsa.pub`: 
   `sshpass -p <BMC_PW> ssh <BMC_IP> "rm -r ~/.ssh/id_rsa.pub`

**Comment by whowutwut** [comment]
@gurevichmark So we don't have sshpass.  @cxhong  may have downloaded it...   I think if we really want to leverage more of this utility, we would want to consider shipping this in xcat-deps.  @chenglch any ideas about this?

**Comment by whowutwut** [comment]
Maybe not in the normal OS 
```
[root@fs3 xcat]# yum install sshpass
Loaded plugins: product-id, search-disabled-repos, subscription-manager
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
No package sshpass available.
Error: Nothing to do
```

**Comment by chenglch** [comment]
For redhat , `sshpass` is shipped in epel repo,  I think we can download the rpms from epel repo for different systems and upload them to GSA directly.

@neo954, is this the correct way to avoid of the conflict like #3202

**Comment by gurevichmark** [comment]
There might be a simpler solution. Instead of `sshpass` utility I can use existing xCAT method:
`my $output = xCAT::RShellAPI::run_remote_shell_api($node, $username, $passwd, $telnet, $verbose, @ARGV[ 1 .. $#ARGV ]);`

The usage example is in `xCAT-server/sbin/rshell_api` tool.

**Comment by whowutwut** [comment]
Yea, for now let's just take the easiest/quickest implementation path that doesn't pull in dependencies, then revisit this when the REST command is provided.

**Comment by whowutwut** [comment]
@gurevichmark I tried this out, and currently this is still in the `unsupported()` code path, is there a way to expose this option only to the user and block other options? Can we do that fine grain blocking?

**Comment by gurevichmark** [comment]
Implemented by PR #3424

**Comment by zet809** [comment]
Close this task as it had been implemented.
